### PR TITLE
 works with multiple extends, git repositories, branches and path

### PR DESCRIPTION
**Pull request by KiranShakya**
works with multiple extends, git repositories, branches and path

### CODE REVIEW

File: src/dry-package.ts [hunk 1]
```diff
@@ -106,45 +109,77 @@ export class DryPackage {
...
+            let realGitRepoUrl: string = extend;
+            let branch: string = '';
+            const matched: string[] = extend.match('([\\w|\\-|_]+)\\.git');
+            if (matched.length && matched.length === 2) {
+                // assume that the reponame is the packageName
+                packageName = matched[1];
+                const splitExtend: string[] = extend.split(packageName + '.git', 2);
+                if (splitExtend.length > 1) {
+                    packageFile = splitExtend[1];
+                    const matchedbranch: string[] = packageFile.match('^/#([\\w|\\-|_]*)');
```
Comment (dejavu1987) [inline src/dry-package.ts]: forwardSlash in the regEx seems to be a typo? @KiranShakya

**Comment by KiranShakya** [comment]
@itonics-tbeauvais, There can be cases that you have some local package files which need extending. The extends  addressed in package from repo can extend local file, or else it should also be addressed as git url.
`"extends":"git@bitbucket.org:itonics/npm-dry-package-files.git#master/dry-itonics.json"`
This change would make it accept the file from repo and also, it won't be overhead since it checks the local clone first.

File: src/dry-package.ts (hunk 1 above)
Comment (KiranShakya) [reply src/dry-package.ts]: It looks like a typo. I thought I removed it. Thanks for correction.

**Comment by itonics-tbeauvais** [comment]
I have merged into your branch. Please make sure you branch in the future, we do not need you to fork. Cheers!

**Pull request merged**